Update wallet-instance-attestation.rst (#50)
italia/eidas-it-wallet-docs: docs/en/wallet-instance-attestation.rst

Wallet Instance Attestation
+++++++++++++++++++++++++++

The Wallet Instance Attestation is data containing details about the Wallet Provider, the Wallet Solution, the Wallet Instance, and the device's security level where the Wallet Instance is installed. It generally attests the **authenticity, integrity, security, privacy, and trust** of a specific Wallet Instance. The Wallet Instance Attestation MUST contain the Wallet Instance public key.

General Properties
------------------

The objectives include:

- Ensuring that the Wallet Instance maintains a level of **integrity** that is capable of preventing any manipulation or forgery attempts by unauthorized third parties.
- Having the Wallet Provider issue a certificate of conformity to assure that the previously mentioned security and trust objectives are fulfilled.

> To meet these requirements, it is necessary for each Wallet Instance to issue a certificate of conformity, guaranteeing its security and compliance with the Trust Model.

Reviewer: ```suggestion
To meet these requirements, it is necessary for each Wallet Instance to issue an attestation of conformity, guaranteeing its security and compliance with the Trust Model.
```

Reviewer: I would prevent the term certificate since it may point to X.509 certificates

> This certificate, called the **Wallet Instance Attestation**, must be electronically signed by its issuer.

Reviewer: ```suggestion
This verifiable attestation, called **Wallet Instance Attestation**, must be electronically signed by its issuer.
```

Dynamic view of the components
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

This section describes the format of the Wallet Instance Attestation and how it is issued by the Wallet Provider.

.. figure:: ../../images/dynamic_view_sequence_wallet_instance_attestation.svg
   :name: sequence diagram for Wallet Instance Attestation request
   :alt: The figure shows the sequence diagram for issuing a Wallet Instance Attestation. The steps will be described below.
   :target: https://www.plantuml.com/plantuml/uml/XPB1RzKm3CRl-IlCJY3nn7s7QOZ3118IGi0kkxYDLLcqJd2SLMz_FLvV6r7AnDN-_Fi-ExajXcfr6iEhh3XC24Rf2Kmh1QoMf4uTQGZPLTnpHZ6u-bv8hm0Br7tz7iUH33wAGwMdHJBpFpLVD3roN35p5qA5qusBhtsQZN7a9uBvekMLzo19GUbNfMBlib8X1_PAaUHveeIPJpTpTmrtPDjiNdrW8iE8Xc7kJgvoeyzh1VeaXYmimnyqi7EcyXP-qddnPAN9EruXYJcnsEhdf1yUrqbqC3MjnM3aOgxT5hmZ8NNrWix8MhQcH_zwMGyaIK-U5KwNgRNGB3yeFIF-kZYyBuNKE4a3VRh_5h0tVbpoTRiROLE__Y_eZOTP9W_RyZOpa5GM4YhbA2uy25fLQgrXkmDANDe7OClN7ktbXO-FyJ8jqluYpguDtVJSFc9y42MCPx04gJDa0Q5vz_LkIMATnjy0

-

>  **Message 1**: The User initializes the Wallet Instance. In particular, this process happens after the Wallet Instance installation and after the expiration of the Wallet Instance Attestation is launched and every time the User wants to request or present (disclose) a credential.

Reviewer: ```suggestion
 **Message 1**: The User initializes the Wallet Instance. In particular, this process happens after the Wallet Instance installation and after the expiration of the Wallet Instance Attestation is launched and every time the User wants to request or present a credential.
```

Reviewer: since the attributes in the credential are disclosed, while the entire credential is presented in its original status

- **Message 2-3**: The Wallet Instance obtains metadata about its Wallet Provider. Among these, we also find the list of supported algorithms, public keys, endpoints.

> - **Message 4**: The Wallet Instance verifies that the Wallet Provider is trustworthy by resolving the provider's trust chain up to the anchor

Reviewer: ```suggestion
- **Message 4**: The Wallet Instance verifies that the Wallet Provider is trustworthy by resolving the provider's trust chain up to the Trust Anchor.
```

> - **Message 5-7**: The Wallet Instance instantiates a new key pair on its TEE and requests a ``nonce`` from the Wallet Provider (as a measure against replay attacks).

Reviewer: ```suggestion
- **Message 5-7**: The Wallet Instance creates a new key pair and requests a ``nonce`` from the Wallet Provider (as a measure against replay attacks).
```

Reviewer: ` and requests a ``nonce`` from the Wallet Provider (as a measure against replay attacks)` @grausof this is under discussion, would it be better to move into the related PR for that? (next milestone?)

> - **Message 8**: The Wallet Instance generates a Wallet Instance Attestation Request, in JWS format, signed with the private key associated with the public key for which it wants to obtain the attestation containing the ``nonce`` and all the required parameters.

Reviewer: ```suggestion
- **Message 8**: The Wallet Instance generates a Wallet Instance Attestation Request, in JWS format, signed with the private key associated with the public key for which it wants to obtain the attestation.
```

Reviewer: not required here
